tmk_core/protocol/usb_descriptor.c

```diff
@@ -40,957 +40,950 @@
- * Device Descriptors
- ******************************************************************************/
-const USB_Descriptor_Device_t PROGMEM DeviceDescriptor =
-{
-    .Header                 = {.Size = sizeof(USB_Descriptor_Device_t), .Type = DTYPE_Device},
-
-    .USBSpecification       = VERSION_BCD(1,1,0),
+/*
+ * Device descriptor
+ */
+const USB_Descriptor_Device_t PROGMEM DeviceDescriptor = {
+  .Header = {
+    .Size                   = sizeof(USB_Descriptor_Device_t),
+    .Type                   = DTYPE_Device
+  },
+  .USBSpecification         = VERSION_BCD(1, 1, 0),
 #if VIRTSER_ENABLE
-    .Class                  = USB_CSCP_IADDeviceClass,
-    .SubClass               = USB_CSCP_IADDeviceSubclass,
-    .Protocol               = USB_CSCP_IADDeviceProtocol,
+  .Class                    = USB_CSCP_IADDeviceClass,
+  .SubClass                 = USB_CSCP_IADDeviceSubclass,
+  .Protocol                 = USB_CSCP_IADDeviceProtocol,
 #else
-    .Class                  = USB_CSCP_NoDeviceClass,
-    .SubClass               = USB_CSCP_NoDeviceSubclass,
-    .Protocol               = USB_CSCP_NoDeviceProtocol,
+  .Class                    = USB_CSCP_NoDeviceClass,
+  .SubClass                 = USB_CSCP_NoDeviceSubclass,
+  .Protocol                 = USB_CSCP_NoDeviceProtocol,
 #endif
-
-    .Endpoint0Size          = FIXED_CONTROL_ENDPOINT_SIZE,
-
-    /* specified in config.h */
-    .VendorID               = VENDOR_ID,
-    .ProductID              = PRODUCT_ID,
-    .ReleaseNumber          = DEVICE_VER,
-
-    .ManufacturerStrIndex   = 0x01,
-    .ProductStrIndex        = 0x02,
-    .SerialNumStrIndex      = 0x03,
-
-    .NumberOfConfigurations = FIXED_NUM_CONFIGURATIONS
+  .Endpoint0Size            = FIXED_CONTROL_ENDPOINT_SIZE,
+  // Specified in config.h
+  .VendorID                 = VENDOR_ID,
+  .ProductID                = PRODUCT_ID,
+  .ReleaseNumber            = DEVICE_VER,
+  .ManufacturerStrIndex     = 0x01,
+  .ProductStrIndex          = 0x02,
+  .SerialNumStrIndex        = 0x03,
+  .NumberOfConfigurations   = FIXED_NUM_CONFIGURATIONS
 };
 
-/*******************************************************************************
- * Configuration Descriptors
- ******************************************************************************/
-const USB_Descriptor_Configuration_t PROGMEM ConfigurationDescriptor =
-{
-    .Config =
-        {
-            .Header                 = {.Size = sizeof(USB_Descriptor_Configuration_Header_t), .Type = DTYPE_Configuration},
-
-            .TotalConfigurationSize = sizeof(USB_Descriptor_Configuration_t),
-            .TotalInterfaces        = TOTAL_INTERFACES,
-
-            .ConfigurationNumber    = 1,
-            .ConfigurationStrIndex  = NO_DESCRIPTOR,
-
-            .ConfigAttributes       = (USB_CONFIG_ATTR_RESERVED | USB_CONFIG_ATTR_REMOTEWAKEUP),
+#ifndef USB_MAX_POWER_CONSUMPTION
+  #define USB_MAX_POWER_CONSUMPTION 500
```

Comment:
Is there a reason behind moving this to the middle of the file?

Reply:
I moved it here to be closer to the actual place where it's used. See below where it's referenced in the MaxPowerConsumption member of the configuration descriptor.